src/Models/BaseElement.php

```diff
@@ -679,6 +679,38 @@ public function getAnchor()
         return $this->anchor = $result;
     }
 
+    /**
+     * Get anchors in this block's content.
+     * Used to populate the "anchor on a page" link in the WYSIWYG
+     *
+     * By default, this finds anchors in any HTMLText field on the block, but
+     * this method should be overridden if anchors are provided in other ways
+     * for this block or if not all HTMLText fields for this block are
+     * displayed on the front-end.
+     */
+    public function getAnchorsInContent(): array
+    {
+        $anchors = [$this->getAnchor()];
+        $anchorRegex = "/\\s+(name|id)\\s*=\\s*([\"'])([^\\2\\s>]*?)\\2|\\s+(name|id)\\s*=\\s*([^\"']+)[\\s +>]/im";
```

Comment:
Also would be nice to modify first part to provide ability to search name or id even if they have a spaces before and after.
`\s+(name|id)\s*?=\s*?(\s*?[\"']\s*?)(\s*?[^\2\s>]*?\s*?)\2`